Archive/Draft/RFCxxxx-PS7-Bootstrapper.md

```diff
@@ -0,0 +1,256 @@
+However, auto-updating will be the default to ensure users have the latest version with security updates.
+Configuration would allow for auto-updating, update via prompt, or don't update.
+
+### Admin vs. Non-Admins
+
+If the process is elevated, then the install will be for the system under `$env:ProgramFiles`.
+
+However, the bootstrapper must work for non-admins.
+Otherwise, PS7 is installed only for the current user if the process is non-elevated.
+In this case, PS7 would be installed to `$env:LOCALAPPDATA\Microsoft\PowerShell7` and the user `$env:PATH` would be appended with this path.
+`PowerShell7` as a folder is used instead of just `PowerShell` as that folder is already being used by PS7 for persisting state.
+
+If PS7 was installed by the bootstrapper elevated so that it's available system wide (under `$env:PROGRAMFILES`), and the user
+is non-admin, but the bootstrapper detects a newer version is available, then it should emit a message indicating that a newer
+version is available, but requires elevation or an administrator to install.
+
+>[Note] Need to verify that Squirrel supports admin installs for system-wide access
+
+### Windows Terminal
+
+Windows Terminal is now a default installed app and has code to detect PS7.
+Currently, it will not detect the prescence of `pwsh.exe` bootstrapper if it exists in system32.
+We may submit a PR to update their detection logic so that the bootstrapper is available as the default shell.
+
+### Offline scenarios
+
+Need to look into integration with [Windows ADK (Assessment and Deployment Kit)](https://docs.microsoft.com/en-us/windows-hardware/get-started/)
+to support pre-provisioning Windows images that already include PS7 rather than bootstrapping at runtime where the machine may be on a factory
+floor (WinPE) without internet access.
+
+### Remoting endpoint
+
+By default, neither WinRM nor SSHD will be setup for PS7 remoting.
+So when remoting from one Windows machine into another, it would not connect to PS7 unless the user has manually configured it
+(and thus also installed PS7).
+
+We may consider adding capability to make it easy to setup PS7 remoting for large deployments in the future.
+For example, maybe `pwsh.exe --enable-ssh-endpoint` and `pwsh.exe --enable-winrm-endpoint` as special switches only for the bootstrapper.
+
+## Alternate Proposals and Considerations
+
+### Reducing size
+
+The bootstrapper itself is small, but installation would still carry a full PS7 package.
+Details of reducing the size of PS7 post-install is outside the scope of this specific RFC.
+
+### Auto-update when used non-interactively
+
+Current design is based on auto-update (unless configuration is set not to) even if the usage is via automation.
+For example, remotely or using task scheduler to execute a script.
+A design decision could be made to not auto-update if we know the usage is not interactive,
```

Comment:
Having my automation engine upgrade outside of my patch cycle is kind of scary. I love the work the team has done to add PS7 to WU/MU. I believe that established process is the better method for upgrading versions.